File: CinemaAPI/CinemaAPI/Models/Film.cs
using System.ComponentModel.DataAnnotations;

namespace CinemaAPI.Models
{

    public class Film
    {
        public int Id { get; set; }
        public string Name { get; set; }
        //public Gen Gen { get; set; }
        public List<Actor>? FilmActors { get; set; }
        public string Image { get; set; }
        [Url(ErrorMessage = "Url is not valid")]
        public string Trailer { get; set; }
        //public List<TimeSlot> Program {  get; set; }   
        public string Description { get; set; }
        public DateTime ReleaseDate { get; set; }
        public string AgeRating { get; set; }
        public List<Review>? Reviews{get;set;}
        public TimeSpan Duration { get; set; }
        
        public DateTime StartRunningDate { get; set; }
        public DateTime EndRunningDate { get; set; }
        
        public List<Reservation> Reservations { get; set; }
        public ICollection<Wishlist> Wishlists { get; set; }

    }
}

File: CinemaApp/src/app/app-logic/film/film.ts
import { Actor } from "./actor";

export interface Film {
    id:number;
    name:string;
    actors:Actor[];
    image:string;
    trailer:string;
    // timeSlot:TimeSlot[];
    description:string;
    releaseDate:Date;
    ageRating:string;
    // reviews:Review[];
    duration:number;
    startRunningDate:Date;
    endRunningDate:Date;
    // reservation:Reservation;
    // whishlist:Whishlist;
}


// export function formatDuration(minutes: number): string {
//   const h = Math.floor(minutes / 60);
//   const m = minutes % 60;
//   return `${h}h ${m}min`;
// }
